packages/x-tree-view/src/internals/plugins/useTreeViewLabel/useTreeViewLabel.itemPlugin.ts

```diff
@@ -27,13 +25,41 @@ export const useTreeViewLabelItemPlugin: TreeViewItemPlugin<any> = ({ props }) =
     propsEnhancers: {
       labelInput: ({
         externalEventHandlers,
-      }): UseTreeItem2LabelInputSlotPropsFromItemsReordering => {
+        interactions,
+      }): UseTreeItem2LabelInputSlotPropsFromLabelEditing => {
         const editable = instance.isItemEditable(itemId);
 
         if (!editable) {
           return {};
         }
 
+        const handleKeydown = (
```

Comment:
Move those two callbacks here, that way `useTreeItem2` is totally agnostic of anything regarding this prop
We have some infos about label editing in other slots but nothing too important